<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Project Organization and Management for Genomics: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [ ['$','$'], ['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="manifest" href="../site.webmanifest">
<link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav data"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Data Carpentry" src="../assets/images/data-logo.svg">
</div>
    </div>
    <div class="selector-container ">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav data" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Data Carpentry" src="../assets/images/data-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Project Organization and Management for Genomics
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
      <i role="img" aria-label="search button" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Project Organization and Management for Genomics
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="data.html">Data</a></li>
<li><a class="dropdown-item" href="old-ncbi.html">Old NCBI</a></li>
          </ul>
</li>
      </ul>
</div>
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
<input class="form-control me-2 searchbox" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-outline-success tablet-search-button" type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="search button"></i>
        </button>
      </fieldset>
</form>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Project Organization and Management for Genomics
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress data">
    <div class="progress-bar data" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
  <div id="sidebar-col" class="col-lg-4">
    <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle"><i class="search-icon" data-feather="x" role="img"></i></button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->
      
            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-tidiness.html">1. Data Tidiness</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-project-planning.html">2. Planning for NGS Projects</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-ncbi-sra.html">3. Examining Data on the NCBI SRA Database</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <li><a href="data.html">Data</a></li>
<li><a href="old-ncbi.html">Old NCBI</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="../instructor/aio.html">See all in one page</a>

            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">
            
            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-01-tidiness"><p>Content from <a href="01-tidiness.html">Data Tidiness</a></p>
<hr>
<p> Last updated on 2023-05-02 | 
        
        <a href="https://github.com/datacarpentry/organization-genomics/edit/main/episodes/01-tidiness.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time <i aria-hidden="true" data-feather="clock"></i> 30 minutes </p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right"> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What metadata should I collect?</li>
<li>How should I structure my sequencing data and metadata?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Think about and understand the types of metadata a sequencing
experiment will generate.</li>
<li>Understand the importance of metadata and potential metadata
standards.</li>
<li>Explore common formatting challenges in spreadsheet data.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="introduction"><h2 class="section-heading">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>When we think about the data for a sequencing project, we often start
by thinking about the sequencing data that we get back from the
sequencing center. However, equally or more important is the data you’ve
generated <em>about</em> the sequences before it ever goes to the
sequencing center. This is the data about the data, often called the
metadata. Without the information about what you sequenced, the sequence
data itself is useless.</p>
<div id="discussion" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="discussion" class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion"></a>
</h3>
<div class="callout-content">
<p>With the person next to you, discuss:</p>
<p>What kinds of data and information have you generated before you sent
your DNA/RNA off for sequencing?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">
  Show me the solution
  </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Types of files and information you have generated:</p>
<ul>
<li>Spreadsheet or tabular data with the data from your experiment and
whatever you were measuring for your study.</li>
<li>Lab notebook notes about how you conducted those experiments.</li>
<li>Spreadsheet or tabular data about the samples you sent off for
sequencing. Sequencing centers often have a particular format they need
with the name of the sample, DNA concentration and other
information.</li>
<li>Lab notebook notes about how you prepared the DNA/RNA for sequencing
and what type of sequencing you’re doing, e.g. paired end Illumina
HiSeq. There likely will be other ideas here too. Was this more
information and data than you were expecting?</li>
</ul>
</div>
</div>
</div>
</div>
<p>All of the data and information just discussed can be considered
metadata, i.e. data about the data. We want to follow a few guidelines
for metadata.</p>
</section><section id="notes"><h2 class="section-heading">Notes<a class="anchor" aria-label="anchor" href="#notes"></a>
</h2>
<hr class="half-width">
<p>Notes about your experiment, including how you prepared your samples
for sequencing, should be in your lab notebook, whether that’s a
physical lab notebook or electronic lab notebook. For guidelines on good
lab notebooks, see the Howard Hughes Medical Institute “Making the Right
Moves: A Practical Guide to Scientifıc Management for Postdocs and New
Faculty” section on <a href="https://www.hhmi.org/sites/default/files/Educational%20Materials/Lab%20Management/Making%20the%20Right%20Moves/moves2_ch8.pdf" class="external-link">Data
Management and Laboratory Notebooks</a>.</p>
<p>Ensure to include dates on your lab notebook pages, the samples
themselves, and in any records about those samples. This will help you
correctly associate samples other later. Using dates also helps create
unique identifiers, because even if you process the same sample twice,
you do not usually do it on the same day, or if you do, you’re aware of
it and give them names like A and B.</p>
<div id="unique-identifiers" class="callout callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="unique-identifiers" class="callout-inner">
<h3 class="callout-title">Unique identifiers<a class="anchor" aria-label="anchor" href="#unique-identifiers"></a>
</h3>
<div class="callout-content">
<p>Unique identifiers are a unique name for a sample or set of
sequencing data. They are names for that data that only exist for that
data. Having these unique names makes them much easier to track
later.</p>
</div>
</div>
</div>
</section><section id="data-about-the-experiment"><h2 class="section-heading">Data about the experiment<a class="anchor" aria-label="anchor" href="#data-about-the-experiment"></a>
</h2>
<hr class="half-width">
<p>Data about the experiment is usually collected in spreadsheets, like
Excel.</p>
<p>What type of data to collect depends on your experiment and there are
often guidelines from metadata standards.</p>
<div id="metadata-standards" class="callout callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="metadata-standards" class="callout-inner">
<h3 class="callout-title">Metadata standards<a class="anchor" aria-label="anchor" href="#metadata-standards"></a>
</h3>
<div class="callout-content">
<p>Many fields have particular ways that they structure their metadata
so it’s consistent and can be used across the field.</p>
<p>The Digital Curation Center maintains <a href="https://www.dcc.ac.uk/resources/metadata-standards/list" class="external-link">a list of
metadata standards</a> and some that are particularly relevant for
genomics data are available from the <a href="https://www.gensc.org/pages/projects.html" class="external-link">Genomics Standards
Consortium</a>.</p>
<p>If there are not metadata standards already, you can think about what
the minimum amount of information is that someone would need to know
about your data to be able to work with it, without talking to you.</p>
</div>
</div>
</div>
</section><section id="structuring-data-in-spreadsheets"><h2 class="section-heading">Structuring data in spreadsheets<a class="anchor" aria-label="anchor" href="#structuring-data-in-spreadsheets"></a>
</h2>
<hr class="half-width">
<p>Regardless of the type of data you’re collecting, there are standard
ways to enter that data into the spreadsheet to make it easier to
analyze later. We often enter data in a way that makes it easy for us as
humans to read and work with it, because we’re human! Computers need
data structured in a way that they can use it. So to use this data in a
computational workflow, we need to think like computers when we use
spreadsheets.</p>
<p>The cardinal rules of using spreadsheet programs for data:</p>
<ul>
<li>Leave the raw data raw - do not change it!</li>
<li>Put each observation or sample in its own row.</li>
<li>Put all your variables in columns - the thing that vary between
samples, like ‘strain’ or ‘DNA-concentration’.</li>
<li>Have column names be explanatory, but without spaces. Use ‘-’, ‘_’
or <a href="https://en.wikipedia.org/wiki/Camel_case" class="external-link">camel case</a>
instead of a space. For instance ‘library-prep-method’ or
‘LibraryPrep’is better than ’library preparation method’ or ‘prep’,
because computers interpret spaces in particular ways.</li>
<li>Do not combine multiple pieces of information in one cell. Sometimes
it just seems like one thing, but think if that’s the only way you’ll
want to be able to use or sort that data. For example, instead of having
a column with species and strain name (e.g. <em>E. coli</em> K12) you
would have one column with the species name (<em>E. coli</em>) and
another with the strain name (K12). Depending on the type of analysis
you want to do, you may even separate the genus and species names into
distinct columns.</li>
<li>Export the cleaned data to a text-based format like CSV
(comma-separated values) format. This ensures that anyone can use the
data, and is required by most data repositories.</li>
</ul>
<p><a href="../files/Ecoli_metadata_composite_messy.xlsx"><img src="../fig/01_tidiness_datasheet_example_messy.png" alt="Messy spreadsheet" class="figure"></a></p>
<div id="discussion-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="discussion-1" class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion-1"></a>
</h3>
<div class="callout-content">
<p>This is some potential spreadsheet data generated about a sequencing
experiment. With the person next to you, for about 2 minutes, discuss
some of the problems with the spreadsheet data shown above. You can look
at the image, or download the file to your computer via this <a href="../files/Ecoli_metadata_composite_messy.xlsx">link</a> and open it in
a spreadsheet reader like Excel.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">
  Show me the solution
  </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>A full set of types of issues with spreadsheet data is at the <a href="https://www.datacarpentry.org/spreadsheet-ecology-lesson/02-common-mistakes/" class="external-link">Data
Carpentry Ecology spreadsheet lesson</a>. Not all are present in this
example. Discuss with the group what they found. Some problems include
not all data sets having the same columns, datasets split into their own
tables, color to encode information, different column names, spaces in
some columns names. Here is a “clean” version of the same
spreadsheet:</p>
<p><a href="https://raw.githubusercontent.com/datacarpentry/wrangling-genomics/gh-pages/files/Ecoli_metadata_composite.tsv" class="external-link">Cleaned
spreadsheet</a> Download the file using right-click (PC)/command-click
(Mac).</p>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="further-notes-on-data-tidiness">Further notes on data tidiness<a class="anchor" aria-label="anchor" href="#further-notes-on-data-tidiness"></a>
</h3>
<p>Organizing your data properly at this point of your experiment will
help your analysis later. It will also prepare your data and notes for
data deposition, which is often required by journals and funding
agencies. If this is a collaborative project, as most projects are now,
it’s also vital information for your collaborators. Well organized data
is very useful for communication and efficiency.</p>
<p>Fear not! If you have already started your project and it’s not set
up this way, there are still opportunities to make updates. One of the
biggest challenges is tabular data that is not formatted so computers
can use it, or has inconsistencies that make it hard to analyze.</p>
<p>More practice on how to structure data is outlined in our <a href="https://www.datacarpentry.org/spreadsheet-ecology-lesson/02-common-mistakes/" class="external-link">Data
Carpentry Ecology spreadsheet lesson</a></p>
<p>Tools like <a href="https://www.datacarpentry.org/OpenRefine-ecology-lesson/" class="external-link">OpenRefine</a>
can help you clean your data.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Keypoints<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Metadata is key for you and others to be able to work with your
data.</li>
<li>Tabular data needs to be structured to be able to work with it
effectively.</li>
</ul>
</div>
</div>
</div>
</div>
</section></section><section id="aio-02-project-planning"><p>Content from <a href="02-project-planning.html">Planning for NGS Projects</a></p>
<hr>
<p> Last updated on 2023-05-02 | 
        
        <a href="https://github.com/datacarpentry/organization-genomics/edit/main/episodes/02-project-planning.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time <i aria-hidden="true" data-feather="clock"></i> 30 minutes </p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right"> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I plan and organize a genome sequencing project?</li>
<li>What information does a sequencing facility need?</li>
<li>What are the guidelines for data storage?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the data we send to and get back from a sequencing
center.</li>
<li>Make decisions about how (if) data will be stored, archived, shared,
etc.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>There are a variety of ways to work with a large sequencing dataset.
You may be a novice who has not used bioinformatics tools beyond doing
BLAST searches. You may have bioinformatics experience with other types
of data and are working with high-throughput (NGS) sequence data for the
first time. In the most important ways, the methods and approaches we
need in bioinformatics are the same ones we need at the bench or in the
field - <em>planning, documenting, and organizing</em> are the key to
good reproducible science.</p>
<div id="discussion" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="discussion" class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion"></a>
</h3>
<div class="callout-content">
<p>Before we go any further, here are some important questions to
consider. If you are learning at a workshop, please discuss these
questions with your neighbor.</p>
<p><strong>Working with sequence data</strong></p>
<p>What challenges do you think you’ll face (or have already faced) in
working with a large sequence dataset? What is your strategy for saving
and sharing your sequence files? How can you be sure that your raw data
have not been unintentionally corrupted? Where/how will you (did you)
analyze your data - what software, what computer(s)?</p>
</div>
</div>
</div>
<section id="sending-samples-to-the-facility"><h2 class="section-heading">Sending samples to the facility<a class="anchor" aria-label="anchor" href="#sending-samples-to-the-facility"></a>
</h2>
<hr class="half-width">
<p>The first step in sending your sample for sequencing will be to
complete a form documenting the metadata for the facility. Take a look
at the following example submission spreadsheet.</p>
<p><a href="../files/sample_submission.txt">Sample submission sheet</a></p>
<p>Download the file using right-click (PC)/command-click (Mac). This is
a tab-delimited text file. Try opening it with Excel or another
spreadsheet program.</p>
<div id="exercise" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise" class="callout-inner">
<h3 class="callout-title">Exercise<a class="anchor" aria-label="anchor" href="#exercise"></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>What are some errors you can spot in the data? Typos, missing data,
inconsistencies?</li>
<li>What improvements could be made to the choices in naming?</li>
<li>What are some errors in the spreadsheet that would be difficult to
spot? Is there any way you can test this?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">
  Show me the solution
  </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Errors:</p>
<ul>
<li>Sequential order of well_position changes</li>
<li>Format of client_sample_id changes and cannot have spaces, slashes,
non-standard ASCII characters</li>
<li>Capitalization of the replicate column changes</li>
<li>Volume and concentration column headers have unusual (not allowed)
characters</li>
<li>Volume, concentration, and RIN column decimal accuracy changes</li>
<li>The prep_date and ship_date formats are different, and prep_date has
multiple formats</li>
<li>Are there others not mentioned?</li>
</ul>
<p>Improvements in naming</p>
<ul>
<li>Shorten client_sample_id names, and maybe just call them “names”
<ul>
<li>For example: “wt” for “wild-type”. Also, they are all “1hr”, so that
is superfluous information</li>
</ul>
</li>
<li>The prep_date and ship_date might not be needed</li>
<li>Use “microliters” for “Volume (µL)” etc.</li>
</ul>
<p>Errors hard to spot:</p>
<ul>
<li>No space between “wild” and “type”, repeated barcode numbers,
missing data, duplicate names</li>
<li>Find by sorting, or counting</li>
</ul>
</div>
</div>
</div>
</div>
</section><section id="retrieving-sample-sequencing-data-from-the-facility"><h2 class="section-heading">Retrieving sample sequencing data from the facility<a class="anchor" aria-label="anchor" href="#retrieving-sample-sequencing-data-from-the-facility"></a>
</h2>
<hr class="half-width">
<p>When the data come back from the sequencing facility, you will
receive some documentation (metadata) as well as the sequence files
themselves. Download and examine the following example file - here
provided as a text file and Excel file:</p>
<ul>
<li><a href="../files/sequencing_results_metadata.txt">Sequencing results -
text</a></li>
<li><a href="../files/sequencing_results_metadata.xls">Sequencing results -
Excel</a></li>
</ul>
<div id="exercise-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1" class="callout-inner">
<h3 class="callout-title">Exercise<a class="anchor" aria-label="anchor" href="#exercise-1"></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>How are these samples organized?</li>
<li>If you wanted to associate the sequence file names with their
corresponding sample names from the submission sheet, could you do so?
How?</li>
<li>What do the _R1/_R2 extensions mean in the file names?</li>
<li>What does the ‘.gz’ extension on the filenames indicate?</li>
<li>What is the total file size - what challenges in downloading and
sharing these data might exist?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">
  Show me the solution
  </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Samples are organized by sample_id</li>
<li>To relate filenames use the sample_id, and do a VLOOKUP on
submission sheet</li>
<li>The _R1/_R2 extensions mean “read 1” and “read 2” of each sample.
These typically refer to forward and reverse reads of the same DNA
fragment from the sequencer, i.e. during paired-end sequencing.</li>
<li>The ‘.gz’ extension means it is a compressed “gzip” type format to
save disk space</li>
<li>The size of all the files combined is 1113.60 Gb (over a terabyte!).
To transfer files this large you should validate the file size following
transfer. Absolute file integrity checks following transfers and methods
for faster file transfers are possible but beyond the scope of this
lesson.</li>
</ol>
</div>
</div>
</div>
</div>
</section><section id="storing-data"><h2 class="section-heading">Storing data<a class="anchor" aria-label="anchor" href="#storing-data"></a>
</h2>
<hr class="half-width">
<p>The raw data you get back from the sequencing center is the
foundation of your sequencing analysis. You need to keep this data, so
that you can always come back to it if there are any questions or you
need to re-run an analysis, or try a new analysis approach.</p>
<div class="section level3">
<h3 id="guidelines-for-storing-data">Guidelines for storing data<a class="anchor" aria-label="anchor" href="#guidelines-for-storing-data"></a>
</h3>
<ul>
<li>Store the data in a place that is accessible by you and other
members of your lab. At a minimum, you and the head of your lab should
have access.</li>
<li>Store the data in a place that is redundantly backed up. It should
be backed up in two locations that are in different physical areas.</li>
<li>Leave the raw data raw. You will be working with this data, but you
do not want to modify this stored copy of the original data. If you
modify the data, you’ll never be able to access those original files. We
will cover how to avoid accidentally changing files in a later lesson in
this workshop <a href="https://datacarpentry.org/shell-genomics/03-working-with-files/#file-permissions" class="external-link">(see
File Permissions)</a>.</li>
</ul>
<div class="section level4">
<h4 id="some-data-storage-solutions">Some data storage solutions<a class="anchor" aria-label="anchor" href="#some-data-storage-solutions"></a>
</h4>
<p>If you have a local high performance computing center or data storage
facility on your campus or with your organization, those are ideal
locations. Get in touch with the people who support those facilities to
ask for information.</p>
<p>If you do not have access to these resources, you can back up on hard
drives. Have two backups, and keep the hard drives in different physical
locations.</p>
<p>You can also use resources like <a href="https://aws.amazon.com/s3/" class="external-link">Amazon S3</a>, <a href="https://azure.microsoft.com/en-us/pricing/details/storage/blobs/" class="external-link">Microsoft
Azure</a>, <a href="https://cloud.google.com/storage/" class="external-link">Google Cloud</a>
or others for cloud storage. The <a href="https://osf.io" class="external-link">open science
framework</a> is a free option for storing files up to 5 GB. See more in
the lesson <a href="https://www.datacarpentry.org/cloud-genomics/04-which-cloud/" class="external-link">“Introduction
to Cloud Computing for Genomics”</a>.</p>
</div>
</div>
</section><section id="summary"><h2 class="section-heading">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<p>Before analysis of data has begun, there are already many potential
areas for errors and omissions. Keeping organized and keeping a critical
eye can help catch mistakes.</p>
<p>One of Data Carpentry’s goals is to help you achieve
<em>competency</em> in working with bioinformatics. This means that you
can accomplish routine tasks, under normal conditions, in an acceptable
amount of time. While an expert might be able to get to a solution on
instinct alone - taking your time, using Google or another Internet
search engine, and asking for help are all valid ways of solving your
problems. As you complete the lessons you’ll be able to use all of those
methods more efficiently.</p>
<div id="where-to-go-from-here" class="callout callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="where-to-go-from-here" class="callout-inner">
<h3 class="callout-title">Where to go from here?<a class="anchor" aria-label="anchor" href="#where-to-go-from-here"></a>
</h3>
<div class="callout-content">
<p>More reading about core competencies</p>
<p>L. Welch, F. Lewitter, R. Schwartz, C. Brooksbank, P. Radivojac, B.
Gaeta and M. Schneider, ‘<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3945096/" class="external-link">Bioinformatics
Curriculum Guidelines: Toward a Definition of Core Competencies</a>’,
PLoS Comput Biol, vol. 10, no. 3, p. e1003496, 2014.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Keypoints<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Data being sent to a sequencing center also needs to be structured
so you can use it.</li>
<li>Raw sequencing data should be kept raw somewhere, so you can always
go back to the original files.</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-03-ncbi-sra"><p>Content from <a href="03-ncbi-sra.html">Examining Data on the NCBI SRA Database</a></p>
<hr>
<p> Last updated on 2023-05-02 | 
        
        <a href="https://github.com/datacarpentry/organization-genomics/edit/main/episodes/03-ncbi-sra.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time <i aria-hidden="true" data-feather="clock"></i> 30 minutes </p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right"> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I access public sequencing data?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Be aware that public genomic data is available.</li>
<li>Understand how to access and download this data.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>In our experiments we usually think about generating our own
sequencing data. However, almost all analyses use reference data, and
you may want to use it to compare your results or annotate your data
with publicly available data. You may also want to do a full project or
set of analyses using publicly available data. This data is a great, and
essential, resource for genomic data analysis.</p>
<p>When you come to publish a paper including your sequencing data, most
journals and funders require that you place your data on a public
repository. Sharing your data makes it more likely that your work will
be re-used and cited. It helps to prepare for this early!</p>
<p>There are many repositories for public data. Some model organisms or
fields have specific databases, and there are ones for particular types
of data. Two of the most comprehensive public repositories are provided
by the <a href="https://www.ncbi.nlm.nih.gov" class="external-link">National Center for
Biotechnology Information (NCBI)</a> and the <a href="https://www.ebi.ac.uk/" class="external-link">European Bioinformatics Institute
(EMBL-EBI)</a>. The NCBI’s <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/" class="external-link">Sequence Read Archive
(SRA)</a> is the database we will be using for this lesson, but the
EMBL-EBI’s Nucleotide Archive (ENA) is also useful. The general
processes are similar for any database.</p>
<section id="accessing-the-original-archived-data"><h2 class="section-heading">Accessing the original archived data<a class="anchor" aria-label="anchor" href="#accessing-the-original-archived-data"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://www.datacarpentry.org/organization-genomics/data/" class="external-link">sequencing
dataset (from Tenaillon, <em>et al.</em> 2016) adapted for this
lesson</a> was obtained from the <a href="https://www.ncbi.nlm.nih.gov/sra" class="external-link">NCBI Sequence Read Archive</a>,
which is a large (~27 petabasepairs/2.7 x 10^16 basepairs as of April
2019) repository for next-generation sequence data. Like many NCBI
databases, it is complex and mastering its use is greater than the scope
of this lesson. Very often there will be a direct link (perhaps in the
supplemental information) to where the SRA dataset can be found. We are
only using a small part of these data, so a direct link cannot be found.
If you have time, go through the following detailed description of
finding the data we are using today (otherwise skip to the next
section).</p>
<div class="section level3">
<h3 id="locate-the-run-selector-table-for-the-lenski-dataset-on-the-sra">Locate the Run Selector Table for the Lenski Dataset on the SRA<a class="anchor" aria-label="anchor" href="#locate-the-run-selector-table-for-the-lenski-dataset-on-the-sra"></a>
</h3>
<p>See the figures below for how information about data access is
provided within the original paper.</p>
<figure><img style="border:1px solid #000000" src="../fig/03_paper_header.png" alt="Screenshot of header from Tenaillon et al. paper (linked in References)" width="800" class="figure mx-auto d-block"></figure><p>The <strong>above image</strong> shows the title of the study, as
well as the authors.</p>
<p>The excerpt from the paper below includes information on how to
locate the sequence data. In this case, the text appears just before the
reference section.</p>
<blockquote>
<p><strong>Author Information</strong> All sequencing data sets are
available in the NCBI BioProject database under accession number
PRJNA294072. The <em>breseq</em> analysis pipeline is available at
GitHub (<a href="https://github.com/barricklab/breseq/" class="external-link">http://github.com/barricklab/breseq</a>).
Other analysis scripts are available at the Dryad Digital Repository (<a href="https://doi.org/10.5061/dryad.6226d" class="external-link">http://dx.doi.org/10.5061/dryad.6226d</a>).
R.E.L. will make strains available to qualified recipients, subject to a
material transfer agreement. Repreints and permissions information is
available at <a href="https://www.nature.com/reprints" class="external-link">www.nature.com/reprints</a>. The
authors declare no competing financial interests. Readers are welcome to
comment on the online version of the paper. Correspondence and requests
for materials should be addressed to R.E.L. (lenski <em>at</em>
msu.edu)</p>
</blockquote>
<p><strong>At the beginning of this workshop we gave you <a href="https://www.datacarpentry.org/organization-genomics/data/" class="external-link">experimental
information about these data</a>. This lesson uses a <em>subset</em> of
SRA files, from a small <em>subproject</em> of the BioProject database
“PRJNA294072”. To find these data you can follow the instructions
below:</strong></p>
<ol style="list-style-type: decimal">
<li><p>Notice that the paper references “PRJNA294072” as a “BioProject”
at NCBI. If you go to the <a href="https://www.ncbi.nlm.nih.gov/" class="external-link">NCBI
website</a> and search for “PRJNA294072” you will be shown a link to the
“Long-Term Evolution Experiment with E. coli” BioProject. Here is the
link to that database: <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA294072" class="external-link">https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA294072</a>.</p></li>
<li><p>Once on the BioProject page, scroll down to the table under
<strong>“This project encompasses the following 15
sub-projects:”</strong>.</p></li>
<li><p>In this table, select <strong>subproject</strong> <em>“<a href="https://www.ncbi.nlm.nih.gov/bioproject/295606" class="external-link">PRJNA295606</a>
SRA or Trace Escherichia coli B str. REL606 E. coli genome evolution
over 50,000 generations (The University of Texas at…)”</em>.</p></li>
<li><p>This will take you to a page with the subproject description, and
a table <strong>“Project Data”</strong> that has a link to the 224 SRA
files for this subproject.</p></li>
<li><p>Click on the number <a href="https://www.ncbi.nlm.nih.gov/sra?linkname=bioproject_sra_all&amp;from_uid=295606" class="external-link">“224”</a>
next to “SRA Experiments” and it will take you to the SRA page for this
subproject. <img src="../fig/03_ncbi_send_results.png" alt="03_send_results.png" class="figure"></p></li>
<li><p>For a more organized table, select “Send results to Run
selector”. This takes you to the Run Selector page for BioProject
PRJNA295606 (the BioProject number for the experiment SRP064605) that is
used in the next section.</p></li>
</ol>
</div>
<div class="section level3">
<h3 id="download-the-lenski-sra-data-from-the-sra-run-selector-table">Download the Lenski SRA data from the SRA Run Selector Table<a class="anchor" aria-label="anchor" href="#download-the-lenski-sra-data-from-the-sra-run-selector-table"></a>
</h3>
<ol style="list-style-type: decimal">
<li><p>Make sure you access the Tenaillon dataset from the provided
link: <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP064605" class="external-link">https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP064605</a>.
This is NCBI’s cloud-based SRA interface. You will be presented with a
page for the overall SRA accession SRP064605 - this is a collection of
all the experimental data.</p></li>
<li><p>Notice on this page there are three sections. “Common Fields”
“Select”, and “Found 312 Items”. Within “Found 312 Items”, click on the
first Run Number (Column “Run” Row “1”). <img src="../fig/03_ncbi_new_tables2.png" alt="ncbi-new-tables2.png" class="figure"></p></li>
<li><p>This will take you to a page that is a run browser. Take a few
minutes to examine some of the descriptions on the page. <img src="../fig/03_ncbi_new_run_browser.png" alt="ncbi-run-browser.png" class="figure"></p></li>
<li><p>Use the browser’s back button to go back to the ‘previous page’.
As shown in the figure below, the second section of the page (“Select”)
has the <strong>Total</strong> row showing you the current number of
“Runs”, “Bytes”, and “Bases” in the dataset to date. On 2022-12-06 there
were 312 runs, 109.58 Gb data, and 177.17 Gbases of data. <img src="../fig/03_ncbi_new_metadata.png" alt="ncbi-new-metadata.png" class="figure"></p></li>
<li><p>Click on the “Metadata” button to download the data for this
lesson. The filename is “SraRunTable.txt” and save it on your computer
Desktop. This text-based file is actually a “comma-delimited” file, so
you should rename the file to “SraRunTable.csv” for your spreadsheet
software to open it correctly.</p></li>
</ol>
<p><strong>You should now have a file called
<code>SraRunTable.csv</code></strong> on your desktop.</p>
<blockquote>
<p>Now you know that comma-separated (and tab-separated) files can be
designated as “text” (<code>.txt</code>) files but use either commas (or
tabs) as <strong>delimiters</strong>, respectively. Sometimes you might
need to use a text-editor (<em>e.g.</em> Notepad) to determine if a file
suffixed with <code>.txt</code> is actually comma-delimited or
tab-delimited.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="review-the-sraruntable-metadata-in-a-spreadsheet-program">Review the SraRunTable metadata in a spreadsheet program<a class="anchor" aria-label="anchor" href="#review-the-sraruntable-metadata-in-a-spreadsheet-program"></a>
</h3>
<p>Using your choice of spreadsheet program, open the
<code>SraRunTable.csv</code> file.</p>
<div id="discussion" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="discussion" class="callout-inner">
<h3 class="callout-title">Discussion<a class="anchor" aria-label="anchor" href="#discussion"></a>
</h3>
<div class="callout-content">
<p>Discuss with the person next to you:</p>
<ol style="list-style-type: decimal">
<li>What strain of <em>E. coli</em> was used in this experiment?</li>
<li>What was the sequencing platform used for this experiment?</li>
<li>What samples in the experiment contain <a href="https://www.illumina.com/technology/next-generation-sequencing/paired-end-sequencing_assay.html" class="external-link">paired
end</a> sequencing data?</li>
<li>What other kind of data is available?</li>
<li>Why are you collecting this kind of information about your
sequencing runs?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">
  Show me the solution
  </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Escherichia coli B str. REL606 shown under the “organism” column.
This is a tricky question because the column labeled “strain” actually
has sample names</li>
<li>The Illumina sequencing platform was used shown in the column
“Platform”. But notice they used multiple instrument types listed under
“Instrument”</li>
<li>Sort by LibraryLayout and the column “DATASTORE_filetype” shows that
“realign,sra,wgmlst_sig” were used for paired-end data, while
“fastq,sra” were used for all single-end reads. (Also notice the
Illumina Genome Analyzer IIx was never used for paired-end
sequencing)</li>
<li>There are several columns including: megabases of sequence per
sample, Assay type, BioSample Model, and more.</li>
<li>These are examples of “metadata” that you should collect for
sequencing projects that are sent to public databases.</li>
</ol>
</div>
</div>
</div>
</div>
<p>After answering the questions, you should avoid saving any changes
you might have made to the metadata file. We do not want to make any
changes. If you were to save this file, make sure you save it as a
text-based <code>.csv</code> file format.</p>
</div>
</section><section id="downloading-a-few-sequencing-files-embl-ebi"><h2 class="section-heading">Downloading a few sequencing files: EMBL-EBI<a class="anchor" aria-label="anchor" href="#downloading-a-few-sequencing-files-embl-ebi"></a>
</h2>
<hr class="half-width">
<p>The SRA does not support direct download of fastq files from its
webpage. However, the <a href="https://www.ebi.ac.uk/ena/browser/home" class="external-link">European Nucleotide
Archive</a> does. Let’s see how we can get a download link to a file we
are interested in.</p>
<ol style="list-style-type: decimal">
<li><p>Navigate to the <a href="https://www.ebi.ac.uk/ena/browser/home" class="external-link">ENA</a>.</p></li>
<li><p>Near the top right, in the box next to “View”, type in
<code>SRR2589044</code> and click the “View” button.</p></li>
<li><p>This will take you to a page with information about the data.
Near the bottom you will have the option to download the data by FTP.
You could download the <code>.fastq</code> read files here, but we do
not need to download these files right now and they are large.
Alternatively, right click and copy the URL to save it for
later.</p></li>
</ol>
<p>We do not recommend downloading large numbers of sequencing files
this way. For that, the NCBI has made a software package called the
<code>sra-toolkit</code>. However, for a couple files, it’s often easier
to go through the ENA.</p>
</section><section id="where-to-learn-more"><h2 class="section-heading">Where to learn more<a class="anchor" aria-label="anchor" href="#where-to-learn-more"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="about-the-sequence-read-archive">About the Sequence Read Archive<a class="anchor" aria-label="anchor" href="#about-the-sequence-read-archive"></a>
</h3>
<ul>
<li>You can learn more about the SRA by reading the <a href="https://www.ncbi.nlm.nih.gov/Traces/sra/" class="external-link">SRA
Documentation</a>
</li>
<li>The best way to transfer a large SRA dataset is by using the <a href="https://www.ncbi.nlm.nih.gov/Traces/sra/?view=toolkit_doc" class="external-link">SRA
Toolkit</a>
</li>
</ul>
</div>
</section><section id="references"><h2 class="section-heading">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<hr class="half-width">
<p>Tenaillon O, Barrick JE, Ribeck N, Deatherage DE, Blanchard JL,
Dasgupta A, Wu GC, Wielgoss S, Cruveiller S, Médigue C, Schneider D,
Lenski RE. Tempo and mode of genome evolution in a 50,000-generation
experiment (2016) Nature. 536(7615): 165–170. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4988878/" class="external-link">Paper</a>,
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4988878/#" class="external-link">Supplemental
materials</a> Data on NCBI SRA: <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP064605" class="external-link">https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP064605</a>
Data on EMBL-EBI ENA: <a href="https://www.ebi.ac.uk/ena/data/view/PRJNA295606" class="external-link">https://www.ebi.ac.uk/ena/data/view/PRJNA295606</a></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Keypoints<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Public data repositories are a great source of genomic data.</li>
<li>You are likely to put your own data on a public repository.</li>
</ul>
</div>
</div>
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
				<p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/datacarpentry/organization-genomics/edit/main/README.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/datacarpentry/organization-genomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a> 
        | <a href="https://github.com/datacarpentry/organization-genomics/" class="external-link">Source</a></p>
				<p><a href="https://github.com/datacarpentry/organization-genomics/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:team@carpentries.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="../LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p><a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">Template licensed under CC-BY 4.0</a> by <a href="https://carpentries.org" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.12.3" class="external-link">sandpaper (0.12.3)</a>,
        <a href="https://github.com/carpentries/pegboard" class="external-link">pegboard (0.5.3)</a>,
      and <a href="https://github.com/carpentries/varnish/tree/0.2.17" class="external-link">varnish (0.2.17)</a>.</p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
			<i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back to top"></i><br><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://datacarpentry.github.io/organization-genomics/instructor/aio.html",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://datacarpentry.github.io/organization-genomics/instructor/aio.html",
  "identifier": "https://datacarpentry.github.io/organization-genomics/instructor/aio.html",
  "dateCreated": "2015-03-24",
  "dateModified": "2023-06-06",
  "datePublished": "2023-06-06"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

